File: jvm/src/main/java/org/karabalin/s1920/Solution1920.java
package org.karabalin.s1920;

import org.junit.jupiter.api.Assertions;

class Solution {
    public int[] buildArray(int[] nums) {
        int[] result = new int[nums.length];
        for (int i = 0; i < nums.length; i++) {
            result[i] = nums[nums[i]];
        }
        return result;
    }
}

public class Solution1920 {
    public static void main(String[] args) {
        var s = new Solution();
        Assertions.assertArrayEquals(new int[]{0, 1, 2, 4, 5, 3}, s.buildArray(new int[]{0, 2, 1, 5, 3, 4}));
    }
}

File: kotlin/src/main/kotlin/org/karabalin/s1920/Solution1920.kt
package org.karabalin.s1920

class Solution {
    fun buildArray(nums: IntArray): IntArray {
        val result = IntArray(nums.size)
        for (i in nums.indices) {
            result[i] = nums[nums[i]]
        }
        return result
    }
}

fun main() {
    val solution = Solution()
    val result = solution.buildArray(intArrayOf(0, 2, 1, 5, 3, 4))
    println(result.joinToString())
}
